fdbclient/ReadYourWrites.actor.cpp

```diff
@@ -1755,10 +1764,68 @@ Future<MappedRangeResult> ReadYourWritesTransaction::getMappedRange(KeySelector
 	}
 
 	Future<MappedRangeResult> result =
-	    reverse ? RYWImpl::readWithConflictRangeForGetMappedRange(
-	                  this, RYWImpl::GetMappedRangeReq<true>(begin, end, mapper, matchIndex, limits), snapshot)
-	            : RYWImpl::readWithConflictRangeForGetMappedRange(
-	                  this, RYWImpl::GetMappedRangeReq<false>(begin, end, mapper, matchIndex, limits), snapshot);
+	    reverse ? RYWImpl::readWithConflictRangeForGetMappedRange<RYWImpl::GetMappedRangeReq, MappedRangeResult>(
+	                  this, RYWImpl::GetMappedRangeReq(begin, end, mapper, limits, true), snapshot)
+	            : RYWImpl::readWithConflictRangeForGetMappedRange<RYWImpl::GetMappedRangeReq, MappedRangeResult>(
+	                  this, RYWImpl::GetMappedRangeReq(begin, end, mapper, limits, false), snapshot);
+
+	return result;
+}
+
+Future<MappedRangeResultV2> ReadYourWritesTransaction::getMappedRangeV2(KeySelector begin,
+                                                                        KeySelector end,
+                                                                        Key mapper,
+                                                                        Key paramsBytes,
+                                                                        GetRangeLimits limits,
+                                                                        Snapshot snapshot,
+                                                                        Reverse reverse) {
+	if (getDatabase()->apiVersionAtLeast(630)) {
+		if (specialKeys.contains(begin.getKey()) && specialKeys.begin <= end.getKey() &&
+		    end.getKey() <= specialKeys.end) {
+			CODE_PROBE(true, "Special key space get range (getMappedRange)", probe::decoration::rare);
+			throw client_invalid_operation(); // Not support special keys.
+		}
+	} else {
+		if (begin.getKey() == "\xff\xff/worker_interfaces"_sr) {
+			throw client_invalid_operation(); // Not support special keys.
+		}
+	}
+	if (checkUsedDuringCommit()) {
```

Comment:
This can be simplified as if the `(!reverse && end < \xff) || (reverse && begin < \xff)`, then throw `client_invalid_operation()`. The idea is that this API is only intended for user key space.